File: src/main/java/ac/dnd/server/admission/infrastructure/web/dto/request/EventCreateRequest.java
package ac.dnd.server.admission.infrastructure.web.dto.request;

import java.time.LocalDateTime;

import ac.dnd.server.common.annotation.StartBeforeEnd;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

@StartBeforeEnd(startDateTime = "startDateTime", endDateTime = "endDateTime")
public record EventCreateRequest(
	@NotBlank
	String name,
	@NotNull
	LocalDateTime startDateTime,
	@NotNull
	LocalDateTime endDateTime
) {
}

File: src/main/kotlin/ac/dnd/server/admission/infrastructure/web/dto/request/EventCreateRequest.kt
package ac.dnd.server.admission.infrastructure.web.dto.request

import ac.dnd.server.shared.validation.StartBeforeEnd
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import java.time.LocalDateTime

@StartBeforeEnd(startDateTime = "startDateTime", endDateTime = "endDateTime")
data class EventCreateRequest(
    @field:NotBlank
    val name: String,

    @field:NotNull
    val startDateTime: LocalDateTime,

    @field:NotNull
    val endDateTime: LocalDateTime
)
